### ISSUE TITLE
MacOS support

### ISSUE THREAD

**Issue by heywoodlh** [MarceColl/zen-browser-flake#13]
I created a zen-browser flake with MacOS support and Linux support: https://github.com/heywoodlh/flakes/tree/main/zen-browser

I think that your Linux support is much better than what I have implemented :smile: -- but it'd be nice if your flake supported MacOS so I could just use your flake!

When I have the willpower, I'll create a Pull Request to implement this in your flake, if desired -- but if you have the time/cycles, MacOS support would be really nice.

Thanks.

**Comment by MarceColl** [comment]
Yes please! I have no Mac hardware to test a flake on so it's not super easy for me to test if everything works, I'd appreciate a contribution with MacOS

**Comment by omarcresp** [comment]
hi @heywoodlh you mind if I port your auto update script and the macos support to this repo?

**Comment by omarcresp** [comment]
I just posted the #50 PR with the auto update script, working on the macOS build from source. Looking for people interested in helping testing it :)

**Comment by Soikr** [comment]
I'm looking forward to this and will gladly test any implemented solution

**Comment by Eveeifyeve** [comment]
> I created a zen-browser flake with MacOS support and Linux support: https://github.com/heywoodlh/flakes/tree/main/zen-browser
> 
> I think that your Linux support is much better than what I have implemented 😄 -- but it'd be nice if your flake supported MacOS so I could just use your flake!
> 
> When I have the willpower, I'll create a Pull Request to implement this in your flake, if desired -- but if you have the time/cycles, MacOS support would be really nice.
> 
> Thanks.

The problem is that the `buildFirefoxMach` isn't available on Darwin and that is because long time ago it was supported by maintainers and then those maintainers went and everything broke so they don't support Darwin anymore because of this.
If you wish zen to be supported on Darwin machines please consider to maintain the `buildFirefoxMach` or at least Firefox on Darwin. Also this is said by Firefox maintainer on nixos [here](https://github.com/NixOS/nixpkgs/pull/347222#issuecomment-2423909603).

**Comment by Eveeifyeve** [comment]
> > I created a zen-browser flake with MacOS support and Linux support: https://github.com/heywoodlh/flakes/tree/main/zen-browser
> > I think that your Linux support is much better than what I have implemented 😄 -- but it'd be nice if your flake supported MacOS so I could just use your flake!
> > When I have the willpower, I'll create a Pull Request to implement this in your flake, if desired -- but if you have the time/cycles, MacOS support would be really nice.
> > Thanks.
> 
> The problem is that the `buildFirefoxMach` isn't available on darwin and that is because long time ago it was supported by maintainers and then those maintainers went and everything broke so they don't support darwin anymore because of this. If you wish zen to be supported on darwin machines please consider to maintain the `buildFirefoxMach` or at least firefox on darwin. Also this is said by firefox maintainer on nixos [here](https://github.com/NixOS/nixpkgs/pull/347222#issuecomment-2423909603).

CC @MarceColl feel free to close this issue as non planned, because it's not possible until someone maintains the Firefox on Darwin machines.
